healthcare/v1/src/test/java/snippets/healthcare/DatasetTests.java

```diff
@@ -63,6 +65,15 @@ public static void checkRequirements() {
     requireEnvVar("GOOGLE_CLOUD_PROJECT");
   }
 
+  @AfterClass
+  public static void deleteTempItems() throws IOException {
+    try {
+      DatasetDelete.datasetDelete(destinationDatasetName);
+    } catch (GoogleJsonResponseException ex) {
+      // Destination dataset already deleted, continue.
```

Comment:
Is this the only case when this exception is thrown? Are there other times this can happen? `GoogleJsonResponseException` seems too generic. Should we check for more specific details to ensure what was the reason for the execption?

Reply:
I think we just need to check for non-2xx responses, which GoogleJsonResponseException does. Any non-2xx response will cause DatasetTests.deleteTempItems to fail.

Reply:
Thank you @noerog . But I am trying to understand if this exception can be thrown for cases where we should not just ignore the exception. 

I am not convinced that the `GoogleJsonResponseException` is thrown only when the said dataset has already been deleted. Correct me if I am wrong. Non-2xx can mean different reasons for the failure. I feel that if someone else updates this test and for some reason a failure is introduced for a reason other than the resource being already deleted, this portion of the test will silently fail always.

Reply:
You're right, it's not only thrown when the dataset has already been deleted, it's thrown whenever there's a non-2xx error code. Do you want to individually check for every possible error that can be returned? If it fails because of something like a quota issue, it'll still fail the test and return an error message stating why it failed. Would changing the comment help?